codex-rs/login/src/token_data.rs

```diff
@@ -17,6 +17,19 @@ pub struct TokenData {
     pub account_id: Option<String>,
 }
 
+impl TokenData {
+    /// Returns true if this is a plan that should use the traditional
+    /// "metered" billing via an API key.
+    pub(crate) fn is_plan_that_should_use_api_key(&self) -> bool {
+        match self.id_token.chatgpt_plan_type.as_deref() {
+            // TODO: Verify this is a comprehensive list of plans that
+            // should NOT use the API key.
+            Some("free") | Some("plus") | Some("pro") | Some("team") => false,
```

Comment:
How about we make plan_type an enum with an Unknown fallback? You can then put `is_plan_that_should_use_api_key` on the enum so a future person is forced to consider this if a new plan type is added.